### ISSUE TITLE
Document main.template.dart

### ISSUE THREAD

**Issue by kwalrath** [dart-lang/site-webdev#1357]
`main.template.dart` is the generated template file that people have to include if they use bootstrapStatic. We should clarify when/how it's generated, why they might see references to it, and when they might need to use it.

This seems similar to the pubspec doc that we should write (#658). Not guide-worthy, but reference-worthy—something easily found and linked to, but unobtrusive.

More info:

* https://github.com/dart-lang/angular/issues/357
* https://github.com/dart-lang/angular/issues/756

**Comment by chalin** [comment]
Of course, I think that this all should be described in the context of Angular AOT compilation/source-to-source translation.  Maybe this is already documented somewhere?

/cc @matanlurey

**Comment by Tomucha** [comment]
> We should clarify when/how it's generated, why they might see references to it, and when they might need to use it.

Amen. Please do. I'm completely baffled.

**Comment by chalin** [comment]
Template file generation is described towards the end of the section on [Configure routes](https://webdev.dartlang.org/angular/tutorial/toh-pt5#configure-routes) of the tutorial, part 5.  I'm thinking that we should probably mention it at the point of first use: the tutorial's starter app. @kwalrath WDYT?

**Comment by Tomucha** [comment]
But when is it generated? While "webdev serve" is running? How does it play with IDE and code completion? And also - explaining WHY would be nice ... I guess it's because of tree-shaking? It deserves some explanation, because importing non-existing files is very confusing (at least to me).

**Comment by Tomucha** [comment]
(to put it into context - I'm trying to upgrade some app from Dart1/Angular2 to Dart2/Angular5 - and it's ... overwhelming)

**Comment by chalin** [comment]
Thanks for your questions! We'll make sure to address them.

**Comment by kwalrath** [comment]
We should mention it early and link to where they can find more information (assuming the explanation isn't trivial).

**Comment by chalin** [comment]
Sounds like the "more info" would be appropriate for an Engineering Note. WDYT? /cc @matanlurey

**Comment by Tomucha** [comment]
Ok, thanks, looking forward to it. My biggest concern is that I will have to manually write all those imports and *NgFactory class names without Idea helping me with code completion. So if you have any tips on how to efficiently work with soon-to-be-generated-but-yet-non-existing-files please include those into the documentation too. Thanks!

**Comment by chalin** [comment]
Yes, that is a known issue. Sorry, I don't have any further guidance to offer. You might try posting to Gitter: https://gitter.im/dart-lang/angular and/or Stackoverflow.
